### DISCUSSION TITLE
error : directus integrate with ldap

### DISCUSSION THREAD

**Discussion by Antonwhy21** [directus/directus#13864] [Archive]
Dear All, 
please help currently I have an issue with the ldap authentication that is giving message on screen : Wrong Username and Password / Invalid user credentials.(directus-out.log)
but if we check on the LDAP server, its allready give success login response, detail as below

**library**
directus: 9.5.1
ldapjs 2.3.1

**config on .env :**
LOG_LEVEL="trace"
AUTH_PROVIDERS="ldap"
AUTH_PROVIDERS_DEFAULT="ldap"
AUTH_LDAP_DRIVER="ldap"
AUTH_LDAP_CLIENT_URL="ldap://ldap.subdomain.domain.io"
AUTH_LDAP_USER_ATTRIBUTE="sAMAccountName"
AUTH_LDAP_BIND_DN="CN=directus 1,OU=Service Account,DC=subdomain,DC=domain,DC=io"
AUTH_LDAP_BIND_PASSWORD="password"
AUTH_LDAP_USER_DN="CN=directus 1,OU=Service Account,DC=subdomain,DC=domain,DC=io"
AUTH_LDAP_GROUP_DN="CN=Directus,OU=Service Account,DC=subdomain,DC=domain,DC=io"

**parameter when creating user on directus**
Role : Directus (same as memberof LDAP & AUTH_LDAP_GROUP_DN="CN=Directus...)
Provider : ldap
External identifier : directus1 (same as username on ldap)

**directus-out.log**
0|directus | 10:45:03 🔍 [1.138ms] select "ip_access" from "directus_roles" where "id" is null limit $1 [1]
0|directus | 10:45:03 🐛 Invalid user credentials.
0|directus | 10:45:03 ✨ request completed POST 401 /auth/login/ldap 37ms
0|directus | 10:45:17 🔍 [1.138ms] select "ip_access" from "directus_roles" where "id" is null limit $1 [1]
0|directus | 10:45:17 🐛 Invalid user credentials.
0|directus | 10:45:17 ✨ request completed POST 401 /auth/login/ldap 18ms

**capture login history from LDAP**
![image](https://user-images.githubusercontent.com/92254163/173467686-821bc133-8a00-4624-ad6b-061da3d9826b.png)

**my question is :**
1. is there any wrong config (.env) or parameter that need to be add when creating users with ldap provider?
2. there is no error log regarding the ldap config, so how can I get more detail on the issue, currently the directus-out.log doesnt give much (only warning Invalid user credentials.).

Thank you for your support :)

**Comment by rijkvanzanten** [comment]
Try setting LOG_STYLE to `raw`. The default formatter might hide away some useful information in this particular case 🤔

**Comment by aidenfoxx** [comment]
The "External Identifier" for LDAP users is their fully qualified DN. You should not have to manually create users for the LDAP login since the provider will create the user for you when they first authenticate.

Your configuration also seems to be incorrect. `AUTH_LDAP_USER_DN` and `AUTH_LDAP_GROUP_DN` should reference the directory containing your user and group records. Something like "OU=Users,DC=ldap,DC=directus,DC=io".

**Reply by Antonwhy21** [reply]
Hi @aidenfoxx 

I allready update based on your info and it give me 503, should I update the **OU** in the ldap server into **Users** instead of **Service Account**

**.env**
AUTH_LDAP_USER_DN="CN=directus 1,OU=Users,DC=ldap,DC=subdomain,DC=domain,DC=io"
AUTH_LDAP_GROUP_DN="CN=Directus,OU=Users,DC=ldap,DC=subdomain,DC=domain,DC=io"
**or**
AUTH_LDAP_USER_DN="CN=directus 1,OU=Users,DC=subdomain,DC=domain,DC=io"
AUTH_LDAP_GROUP_DN="CN=Directus,OU=Users,DC=subdomain,DC=domain,DC=io"

**get error** 
0|directus  | 07:50:12 🔍 [0.894ms] select "ip_access" from "directus_roles" where "id" is null limit $1 [1]
0|directus  | 07:50:12 🐛 Service returned unexpected error
0|directus  | 07:50:12 ✨ request errored POST 503 /auth/login/ldap 43ms

**the OU is based on value DistighuishedName on ldap settings**
DistighuishedName : CN=directus 1,**OU=Service Account**,DC=subdomain,DC=domain,DC=io

**Reply by Antonwhy21** [reply]
allready update the OU : Users in ldap server, and update the AUTH_LDAP_USER_DN & AUTH_LDAP_GROUP_DN  but currently it give error : Invalid bind user
DistighuishedName : CN=directus 1,OU=Users,DC=subdomain,DC=domain,DC=co,DC=id

0|directus  | 09:47:33 🔍 [1.520ms] select "ip_access" from "directus_roles" where "id" is null limit $1 [1]
0|directus  | 09:47:33 🐛 Invalid bind user
0|directus  | 09:47:33 ✨ request errored POST 503 /auth/login/ldap 11ms

**Comment by Antonwhy21** [comment]
Thanks for @aidenfoxx , @rijkvanzanten  for the response, currently the issue already fix, turns out its connection issue, after update the hostname with ip address, its working fine now.
AUTH_LDAP_CLIENT_URL="ldap://10.xxx.xx.xxx:389"
and use other settings in the .env as per my first post.
